hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/CustomCreateOperation.java

```diff
@@ -50,7 +51,7 @@ protected long cost(@NonNull final MessageFrame frame) {
         final var updater = (ProxyWorldUpdater) frame.getWorldUpdater();
 
         final var origin = frame.getRecipientAddress();
-        final var originNonce = updater.getAccount(origin).getNonce();
+        final var originNonce = requireNonNull(updater.getAccount(origin)).getNonce();
         // Decrement nonce by 1 to normalize the effect of transaction execution
         final var address = Address.contractAddress(origin, originNonce - 1);
```

Comment:
How come `CustomCreate2Operation.setupPendingCreation` doesn't have to decrement the nonce?  What's the difference?

Reply:
Because when determining the address where we need to create the new contract, for `Create2` we are given the address and for `Create` we need to calculate the address and part of the address calculation depends on using the correct nonce.